src/workspace.rs

```diff
@@ -411,3 +429,170 @@ impl Workspace {
+}
+
+impl Default for Profile {
+    fn default() -> Profile {
+        Profile {
+            opt_level: OptLevel::One,
+            lto: Lto::Bool(false),
+            codegen_units: None,
+            overflow_checks: false,
+            panic: PanicStrategy::Unwind,
+        }
+    }
+}
+
+impl Profile {
+    /// The preferred set of defaults for compiling a release build of a contract
+    pub fn default_contract_release() -> Profile {
+        Profile {
+            opt_level: OptLevel::Z,
+            lto: Lto::Bool(true),
+            codegen_units: Some(1),
+            overflow_checks: true,
+            panic: PanicStrategy::Abort,
+        }
+    }
+
+    /// Set any unset profile settings from the config.
+    ///
+    /// Therefore:
+    ///   - If the user has explicitly defined a profile setting, it will not be overwritten.
+    ///   - If a profile setting is not defined, the value from this profile instance will be added
+    fn merge(&self, profile: &mut value::Table) {
+        let mut set_value_if_vacant = |key: &'static str, value: value::Value| {
+            if !profile.contains_key(key) {
+                profile.insert(key.into(), value);
+            }
+        };
+        set_value_if_vacant("opt-level", self.opt_level.to_toml_value());
+        set_value_if_vacant("lto", self.lto.to_toml_value());
+        if let Some(codegen_units) = self.codegen_units {
+            set_value_if_vacant("codegen-units", codegen_units.into());
+        }
+        set_value_if_vacant("overflow-checks", self.overflow_checks.into());
+        set_value_if_vacant("panic", self.panic.to_toml_value());
+    }
+}
+
+/// The [`opt-level`](https://doc.rust-lang.org/cargo/reference/profiles.html#opt-level) setting
+#[derive(Clone, Copy, PartialEq, Eq, Debug, Hash, PartialOrd, Ord)]
+#[allow(unused)]
+pub enum OptLevel {
+    Zero,
+    One,
+    Two,
+    Three,
+    S,
+    Z,
+}
+
+impl OptLevel {
+    fn to_toml_value(&self) -> value::Value {
+        match self {
+            OptLevel::Zero => 0.into(),
+            OptLevel::One => 1.into(),
+            OptLevel::Two => 2.into(),
+            OptLevel::Three => 3.into(),
+            OptLevel::S => "s".into(),
+            OptLevel::Z => "z".into(),
+        }
+    }
+}
+
+/// The [`link-time-optimization`](https://doc.rust-lang.org/cargo/reference/profiles.html#lto) setting.
+#[derive(Clone, Copy, PartialEq, Eq, Debug, Hash, PartialOrd, Ord)]
+pub enum Lto {
+    /// False = no LTO
+    /// True = "Fat" LTO
+    Bool(bool),
+    /// Named LTO settings like "thin".
+    #[allow(unused)]
+    Named(&'static str),
+}
+
+impl Lto {
+    fn to_toml_value(&self) -> value::Value {
+        match self {
+            Lto::Bool(b) => value::Value::Boolean(*b),
+            Lto::Named(n) => value::Value::String(n.to_string()),
+        }
+    }
+}
+
+/// The `panic` setting.
+#[derive(Clone, Copy, PartialEq, Eq, Debug, Hash, PartialOrd, Ord)]
+pub enum PanicStrategy {
+    Unwind,
```

Comment:
Not sure how this is every going to be applicable for Wasm smart contracts to be honest.

Reply:
Or is it?

Reply:
See above, gonna keep it for reference because it is a valid value for `PanicStrategy`, even though unused/not applicable for the wasm contract build itself